lib/Command/ExpirePassword.php

```diff
@@ -106,54 +129,110 @@ protected function configure() {
+			$users = [];
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === false) {
+					$output->writeln("Group name $group doesn't exist");
+					return 1;
+				} else {
+					$users[$group] = [];
+					foreach ($this->getAllUsersFromGroup($group) as $users_array) {
+						$users[$group] = $users_array;
+						$this->setPasswordExpiry($input, $output, $users[$group], $group);
+					}
+				}
+			}
+		} else {
+
+			$uid = $input->getArgument('uid');
+
+			/** @var $user \OCP\IUser */
+			$user = $this->userManager->get($uid);
+
+			if ($user === null) {
+				$output->writeln("<error>Unknown user: $uid</error>");
+				return self::EX_NOUSER;
+			}
+
+			if (!$user->canChangePassword()) {
+				$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
+				return self::EX_GENERAL_ERROR;
+			}
+
+			if ($user !== null) {
+				$this->setPasswordExpiry($input, $output, [$user]);
+			}
 		}
 
-		$expireDate = new \DateTime();
-		$expireDate->setTimezone(new \DateTimeZone('UTC'));
-		$expireDate->setTimestamp($this->timeFactory->getTime());
-		$expireDate->modify($input->getArgument('expiredate'));
-
-		$oldDate = new \DateTime();
-		$oldDate->setTimezone(new \DateTimeZone('UTC'));
-		$oldDate->setTimestamp($this->timeFactory->getTime());
-		$oldDate->modify($input->getArgument('expiredate'));
+		return 0;
+	}
 
-		if ($this->config->getAppValue('password_policy', 'spv_user_password_expiration_checked', false) === 'on') {
-			$delta = $this->config->getAppValue('password_policy', 'spv_user_password_expiration_value', 90);
-			$oldDate->modify("-$delta days");
+	protected function getAllUsersFromGroup($group) {
+		$count = 0;
+		$users = [];
+		foreach ($this->groupManager->findUsersInGroup($group) as $user) {
+			\array_push($users, $user);
+			$count++;
+			//Take 200 users at a time
+			if ($count > 199) {
+				yield $users;
+				$count = 1;
+				$users = [];
+			}
+		}
+		if (\count($users) > 0) {
+			yield $users;
 		}
+	}
 
-		$this->config->deleteUserValue(
-			$uid,
-			'password_policy',
-			'forcePasswordChange'
-		);
-
-		// add a dummy password in the user_password_history so the cron job
-		// can notify about the expiration of the password.
-		$oldPassword = new OldPassword();
-		$oldPassword->setUid($uid);
-		$oldPassword->setPassword(OldPassword::EXPIRED);
-		$oldPassword->setChangeTime($oldDate->getTimestamp());
-		$this->mapper->insert($oldPassword);
-
-		// show expire date if it was given
-		if ($input->hasArgument('expiredate')) {
-			$output->writeln("The password for $uid is set to expire on ". $expireDate->format('Y-m-d H:i:s T').'.');
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @param IUser[] $users
+	 */
+	protected function setPasswordExpiry(InputInterface $input, OutputInterface $output, $users) {
+		foreach ($users as $user) {
+			$expireDate = new \DateTime();
```

Comment:
feels a bit overkill to reinitialize everything from the command in this method. this method has too many responsibilities. In my view, this method should only be responsible of setting the expiration for a list of users, not doing anything with $input or $output.

this this current code, you'd set the timestamp to a different time for every user as you iterate... I'd prefer consistency here

Reply:
I have updated the code. Wrote another method which would call `setPasswordExpiry`. The reinitialise of `expireDate`  and `oldDate` are now taken care once by the new function. Hence removed the usage of `input` and `output` from `setPasswordExpiry()` method.